File: src/main/java/socialmedia/server/post/Post.java
package socialmedia.server.post;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import socialmedia.server.user.User;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String title;

    private String content;

    @CreationTimestamp
    @Column(updatable = false, nullable = false)
    private Date createdOn;

    private Status status;

    @ManyToOne
    @JoinColumn(name="user_id", nullable = true)
    private User user;
}

File: SocialMediaDesktopClient/SocialMediaDesktopClient/Post.cs
using System;
using SocialMediaClient;




public class Post
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Content { get; set; }
    public int UserId { get; set; }
    public DateTime CreatedOn { get; set; }
    public PostStatus Status { get; set; }

    public override string ToString()
    {
        return $"Title: {Title}, Content: {Content}, Created On: {CreatedOn}";
    }
    public enum PostStatus
    {
        PENDING,
        PUBLISHED,
        REMOVED
    }

}

